packages/story-editor/src/components/panels/design/textAccessibility/textAccessibility.js

```diff
@@ -0,0 +1,167 @@
+import { MULTIPLE_VALUE, MULTIPLE_DISPLAY_VALUE } from '../../../../constants';
+import { combineElementsWithTags } from './utils';
+
+// Options to be mapped for the dropdown
+const semanticHeadingsOptionsMap = [
+  {
+    label: __('Heading 1', 'web-stories'),
+    value: 'h1',
+  },
+  {
+    label: __('Heading 2', 'web-stories'),
+    value: 'h2',
+  },
+  {
+    label: __('Heading 3', 'web-stories'),
+    value: 'h3',
+  },
+  {
+    label: __('Paragraph', 'web-stories'),
+    value: 'p',
+  },
+  {
+    label: __('Automatic', 'web-stories'),
+    value: 'auto',
+  },
+];
+
+function TextAccessibilityPanel({ selectedElements, pushUpdate }) {
+  // Feature flagging for Semantic Headings
+  const showSemanticHeadings = useFeature('showSemanticHeadings');
+  const [selectedTextElements, setSelectedTextElements] = useState([]);
+  const [currentTags, setCurrentTags] = useState([]);
+  useEffect(() => {
+    // We only want to impact elements if all selected elements are of type text
+    const textElements = selectedElements.filter(({ type }) => 'text' === type);
+    // Then we want to get the text tags for the elements if not already defined
+    const textTags = getTextElementTagNames(textElements);
+    // Then combine the elements with their associated tag
+    const newElements = combineElementsWithTags(textElements, textTags);
+    // And then set them into component state
+    setSelectedTextElements(newElements);
+    setCurrentTags(Array.from(textTags.values()));
+  }, [selectedElements, setSelectedTextElements]);
+  const { highlight, resetHighlight, cancelHighlight } = useHighlights(
+    (state) => ({
+      highlight: state[states.ASSISTIVE_TEXT],
+      resetHighlight: state.onFocusOut,
+      cancelHighlight: state.cancelEffect,
+    })
+  );
+  if (!showSemanticHeadings) {
+    return null;
+  }
+  // Map all types of tag names in the selected elements
+  // and then convert to an Array for usage
+  const handleChange = (ev, value) => {
+    selectedElements.map(() => {
+      return pushUpdate({ tagName: value });
+    });
+    // Update the selected text elements with the
+    // value chosen in the text accessibility dropdown
+    const newTags = getTextElementTagNames(selectedTextElements, value);
+    setCurrentTags(Array.from(newTags.values()));
+  };
+  // Check if tags match for selected item in dropdown
+  // If they don't match, we want to show 'Mixed'
+  // However, if they match, show the matching value
+  const currentValue =
+    new Set(currentTags || []).size === 1 ? currentTags[0] : '((MULTIPLE))';
+  return (
+    <SimplePanel
+      css={highlight && styles.FLASH}
+      onAnimationEnd={() => resetHighlight()}
+      name="textAccessibility"
+      title={__('Accessibility', 'web-stories')}
+      isPersistable={!highlight}
+    >
+      <Row>
+        <DropDown
+          ref={(node) => {
+            if (node && highlight?.focus && highlight?.showEffect) {
```

Comment:
Q: I don't currently see a way in the editor to actually highlight the panel, is this something that will be done in a follow-up issue?

Reply:
@miina This is existing logic from another panel used in this space.

I do see a highlight around the object and I am able to navigate with it using keyboard navigation, as well as the other panels within this space. Are you not seeing this work within the QA instance?

<img width="360" alt="Screen Shot 2022-07-11 at 12 49 17 PM" src="https://user-images.githubusercontent.com/7110244/178316281-d69b1495-077f-4e75-bf74-235851632eae.png">
<img width="367" alt="Screen Shot 2022-07-11 at 12 49 39 PM" src="https://user-images.githubusercontent.com/7110244/178316349-d49397f3-7299-48a3-85ff-5d41b3ce4484.png">